scene/resources/tile_set.cpp

```diff
@@ -5360,13 +5361,31 @@ int TileData::get_y_sort_origin() const {
 
 void TileData::set_occluder(int p_layer_id, Ref<OccluderPolygon2D> p_occluder_polygon) {
 	ERR_FAIL_INDEX(p_layer_id, occluders.size());
-	occluders.write[p_layer_id] = p_occluder_polygon;
+	occluders.write[p_layer_id].occluder = p_occluder_polygon;
+	occluders.write[p_layer_id].transformed_occluders.clear();
 	emit_signal(SNAME("changed"));
 }
 
-Ref<OccluderPolygon2D> TileData::get_occluder(int p_layer_id) const {
+Ref<OccluderPolygon2D> TileData::get_occluder(int p_layer_id, bool p_flip_h, bool p_flip_v, bool p_transpose) const {
 	ERR_FAIL_INDEX_V(p_layer_id, occluders.size(), Ref<OccluderPolygon2D>());
-	return occluders[p_layer_id];
+
+	int key = int(p_flip_h) | int(p_flip_v) << 1 | int(p_transpose) << 2;
+	if (key == 0) {
+		return occluders[p_layer_id].occluder;
+	}
+
+	if (occluders[p_layer_id].occluder.is_null()) {
+		return Ref<OccluderPolygon2D>();
+	}
+
+	if (!occluders[p_layer_id].transformed_occluders.has(key)) {
```

Comment:
```suggestion
	HashMap<int, Ref<OccluderPolygon2D>>::Iterator I = occluders[p_layer_id].transformed_occluders.find(key);
	if (!I) {
```
Avoids double lookup (although you'd need to tweak the logic a bit).